### PR TITLE
Implicit scrolling for pageview

### LINKED ISSUE #44484: PageView should allow for implicit a11y scrolling
`PageView` normally does not allow implicit a11y scrolling, but some applications may want that - e.g. if used to create a carousel type of widget that would otherwise have many of the same behaviors as a page view (snap-to scrolling) but that should otherwise participate in a11y navigation for the current page.

/cc @goderbauer @darrenaustin

### PR DESCRIPTION
**Pull request by dnfield**
## Description

Previous attempt here: https://github.com/flutter/flutter/pull/44645

This is an improvement on that (I hope) in terms of documentation and implmentation.  No longer uses a layout builder.

If we can get the iOS bridge to respect the `allowImplicitScrolling` flag, we could also allow for the page view to have a non-zero cache extent - but for now, on iOS, having a cache extent allows implicit a11y scrolling.

Basic use case here is to use a PageView in non-full-screen applications such as a carousel.

## Related Issues

https://github.com/flutter/flutter/pull/44645
fixes https://github.com/flutter/flutter/issues/44484

## Tests

I added the following tests:

Test asserting the semantics tree looking right when the new flag is set.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/page_view.dart [hunk 1]
```diff
@@ -647,10 +681,25 @@ class PageView extends StatefulWidget {
...
+       assert(allowImplicitScrolling != null),
        controller = controller ?? _defaultPageController,
        super(key: key);
 
+  /// Controls whether the widget's pages will respond to
+  /// [RenderObject.showOnScreen], which will allow for implicit accessibility
+  /// scrolling.
+  ///
+  /// With this flag set to false, when accessibility focus reaches the end of
+  /// the current page and a user performs another implicit scroll action,
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/page_view.dart]: the "implicit scroll action" is moving a11y focus to the next element. Maybe we should say:

With this flag set to false, when accessibility focus reaches the end of current page and the user attempts to move it to the next element, the focus will traverse to the next widget outside of the page view.

With this flag set to true, when accessibility focus reaches the end of the current page and user attempts to move it to the next element, focus will traverse to the next page in the page view.

File: packages/flutter/lib/src/widgets/page_view.dart [hunk 2]
```diff
@@ -754,7 +805,8 @@ class _PageViewState extends State<PageView> {
         physics: physics,
         viewportBuilder: (BuildContext context, ViewportOffset position) {
           return Viewport(
-            cacheExtent: 0.0,
+            cacheExtent: widget.allowImplicitScrolling ? 1.0 : 0.0,
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/page_view.dart]: Maybe leave a TODO here to make the cacheExtend/cacheExtendStyle configurable on the pageView once we have implicitScrolling set up correctly on iOS.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/page_view.dart (hunk 2 above)
Comment (dnfield) [reply packages/flutter/lib/src/widgets/page_view.dart]: Done

File: packages/flutter/lib/src/widgets/page_view.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter/lib/src/widgets/page_view.dart]: I like that better, thanks - done.

**Comment by dnfield** [comment]
The commit behind this one is about to be reverted while I investigate perf regression that started with it.  Will have to put this on hold.

**Comment by dnfield** [comment]
Regression was unrelated, code is safe for now

**Pull request merged**